In [75]:
import numpy as np
import pandas as pd

In [76]:
df = pd.read_csv('./data/penguins_lter.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 17 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   studyName            344 non-null    object 
 1   Sample Number        344 non-null    int64  
 2   Species              344 non-null    object 
 3   Region               344 non-null    object 
 4   Island               344 non-null    object 
 5   Stage                344 non-null    object 
 6   Individual ID        344 non-null    object 
 7   Clutch Completion    344 non-null    object 
 8   Date Egg             344 non-null    object 
 9   Culmen Length (mm)   342 non-null    float64
 10  Culmen Depth (mm)    342 non-null    float64
 11  Flipper Length (mm)  342 non-null    float64
 12  Body Mass (g)        342 non-null    float64
 13  Sex                  334 non-null    object 
 14  Delta 15 N (o/oo)    330 non-null    float64
 15  Delta 13 C (o/oo)    331 non-null    flo

Dátová sada obsahuje 344 riadkov a 17 stĺpcov.
## Doplnenie chýbajúcich hodnôt
Zistime počet chýbajúcich hodnôt.

In [77]:
# pocet
df.isna().sum()

# percentaz chybajucich hodnot
# df.isnull().sum() * 100 / df.shape[0]

studyName                0
Sample Number            0
Species                  0
Region                   0
Island                   0
Stage                    0
Individual ID            0
Clutch Completion        0
Date Egg                 0
Culmen Length (mm)       2
Culmen Depth (mm)        2
Flipper Length (mm)      2
Body Mass (g)            2
Sex                     10
Delta 15 N (o/oo)       14
Delta 13 C (o/oo)       13
Comments               318
dtype: int64

Z vypísania dátovej sady sme zistili, že dve vzorky tučniakov (3, 339) neobsahujú väčšinu parametrov. Jedna z ponzámok napovedá, že dospelý jedinec nebol  odmeraný, tak tieto chybné vzorky odstránime. Doplňme ostatné chýbajúce číselné hodnoty vzoriek z krvi (Delta 15, Delta 13) ich strednou hodnotou. Percentuálne tieto chýbajúce hodnoty zaberajú ~3.5 % miesta v dátovej sade. Ďalej už len ostáva doplniť chýbajúce pohlavia tučniakov (~2 %) najčastejšou vyskytovanou hodnotou v sade. Taktiež vzorka tučniaka na riadku s indexom 336 obsahuje chybnú hodnotu (`.`) pohlavia. Nahradíme ju modusom tiež.

In [78]:
# odstranenie chybajucich vzoriek
bad_samples = [3, 339]
for x in bad_samples: df.drop(x, axis=0, inplace=True, errors = 'ignore')

# doplnenie vzoriek z krvi ich strednou hodnotou
missing_values = ['Delta 15 N (o/oo)', 'Delta 13 C (o/oo)']
for x in missing_values : df[x].fillna(value = df[x].mean(), inplace=True)

# doplnenie pohlavia modusom
df['Sex'].fillna(df['Sex'].mode().values[0], inplace=True)
df.loc[336,'Sex'] = 'MALE'
df.isna().sum()

studyName                0
Sample Number            0
Species                  0
Region                   0
Island                   0
Stage                    0
Individual ID            0
Clutch Completion        0
Date Egg                 0
Culmen Length (mm)       0
Culmen Depth (mm)        0
Flipper Length (mm)      0
Body Mass (g)            0
Sex                      0
Delta 15 N (o/oo)        0
Delta 13 C (o/oo)        0
Comments               317
dtype: int64

## Odstránenie nepotrebných hodnôt
Z dátovej sady môžeme zmazať identifikačné hodnoty, ktorých kombinácia len identifikuje konkrétneho tučniaka - Individual ID, Sample Number.
Komentáre k odberu vzoriek sú taktiež nepotrebné na ďalšiu analýzu.

In [79]:
df.drop(['Individual ID', "Sample Number", "Comments"], axis = 1, inplace=True, errors = 'ignore')